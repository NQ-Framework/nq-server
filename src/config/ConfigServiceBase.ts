export class ConfigServiceBase {
    protected throwError(param: string): never {
        throw new Error(`missing configuration item: ${param}`);
    }
}