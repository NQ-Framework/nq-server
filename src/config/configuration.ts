export default () => ({
  port: parseInt(process.env.PORT as string, 10) || 80,
  auth: {
    clientEmail: process.env.CLIENT_EMAIL,
    privateKey: process.env.PRIVATE_KEY,
    projectId: process.env.PROJECT_ID,
  },
  db: {},
});
