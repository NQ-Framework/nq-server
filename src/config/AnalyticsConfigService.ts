import { Injectable } from '@nestjs/common';
import { ConfigService } from '@nestjs/config';
import { ConfigServiceBase } from './ConfigServiceBase';

@Injectable()
export class AnalyticsConfigService extends ConfigServiceBase {
  constructor(private config: ConfigService) {
    super();
  }

  get measurementId(): string {
    return (
      this.config.get('analytics.measurementId') ??
      this.throwError('measurement id')
    );
  }
  get apiSecret(): string {
    return (
      this.config.get('analytics.apiSecret') ?? this.throwError('api secret')
    );
  }
}
