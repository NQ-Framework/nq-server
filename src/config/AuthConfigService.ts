import { Injectable } from '@nestjs/common';
import { ConfigService } from '@nestjs/config';

@Injectable()
export class AuthConfigService {
  constructor(private config: ConfigService) {}

  get projectId(): string {
    return this.config.get('auth.projectId') ?? this.throwError('project id');
  }
  get privateKey(): string {
    return this.config.get('auth.privateKey') ?? this.throwError('private key');
  }
  get clientEmail(): string {
    return this.config.get('auth.clientEmail') ?? this.throwError('cl email');
  }
  private throwError(param: string): never {
    throw new Error(`missing configuration item: ${param}`);
  }
}
