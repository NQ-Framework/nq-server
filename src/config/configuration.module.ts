import { Module } from '@nestjs/common';
import { AuthConfigService } from './AuthConfigService';

@Module({
  providers: [AuthConfigService],
  exports: [AuthConfigService],
})
export class ConfiugrationModule {}
