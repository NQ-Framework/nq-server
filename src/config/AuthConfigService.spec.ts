import { Test, TestingModule } from '@nestjs/testing';
import { AuthConfigService } from './AuthConfigService';
import { ConfigService } from '@nestjs/config';

describe('AuthConfigService', () => {
    let service: AuthConfigService;

    const mockConfigService = {
        get: jest.fn()
    }

    beforeEach(async () => {
        const module: TestingModule = await Test.createTestingModule({
            providers: [AuthConfigService, { provide: ConfigService, useValue: mockConfigService }],
        }).compile();

        service = module.get<AuthConfigService>(AuthConfigService);
    });

    it('should be defined', () => {
        expect(service).toBeDefined();
    });
    it('should get settings from config service', () => {
        mockConfigService.get.mockReturnValue("test value");
        expect(service.projectId).toEqual("test value");
        expect(service.privateKey).toEqual("test value");
        expect(service.clientEmail).toEqual("test value");
    });

});
