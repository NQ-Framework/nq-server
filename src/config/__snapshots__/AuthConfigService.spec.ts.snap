// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthConfigService should throw on missing configuration 1`] = `"missing configuration item: project id"`;

exports[`AuthConfigService should throw on missing configuration 2`] = `"missing configuration item: private key"`;

exports[`AuthConfigService should throw on missing configuration 3`] = `"missing configuration item: cl email"`;
